### ISSUE TITLE
Panic during shutdown of server

### ISSUE THREAD

**Issue by e-dard** [influxdata/influxdb#8170]
During a shutdown using the `inmem` index:

```
^C[I] 2017-03-20T20:09:19Z Signal received, initializing clean shutdown...
[I] 2017-03-20T20:09:19Z Waiting for clean shutdown...
[I] 2017-03-20T20:09:19Z snapshot listener closed service=snapshot
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xe8 pc=0x82528f]

goroutine 77 [running]:
github.com/influxdata/influxdb/tsdb.(*Shard).SeriesSketches(0xc42016a9c0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/edd/Development/golang/src/github.com/influxdata/influxdb/tsdb/shard.go:495 +0xaf
github.com/influxdata/influxdb/tsdb.(*Store).SeriesCardinality.func1(0xc42016a9c0, 0xc44c74af20, 0xc44c738140, 0x1, 0x1, 0xc44c736ff0, 0x0)
	/Users/edd/Development/golang/src/github.com/influxdata/influxdb/tsdb/store.go:717 +0xcc
github.com/influxdata/influxdb/tsdb.(*Store).estimateCardinality(0xc42016e280, 0xc4202c040c, 0x6, 0xc372a8, 0xf6e910, 0xc4200fa1d8, 0xc44c736ff0)
	/Users/edd/Development/golang/src/github.com/influxdata/influxdb/tsdb/store.go:691 +0x101
github.com/influxdata/influxdb/tsdb.(*Store).SeriesCardinality(0xc42016e280, 0xc4202c040c, 0x6, 0xc44c746ab8, 0xc44c73aed0, 0x0)
	/Users/edd/Development/golang/src/github.com/influxdata/influxdb/tsdb/store.go:718 +0x4b
github.com/influxdata/influxdb/tsdb.(*Store).Statistics(0xc42016e280, 0xc420261200, 0x1, 0x1, 0xc44c74ad20)
	/Users/edd/Development/golang/src/github.com/influxdata/influxdb/tsdb/store.go:95 +0x13e
github.com/influxdata/influxdb/cmd/influxd/run.(*Server).Statistics(0xc42016e140, 0xc420261200, 0xc420180140, 0xc4200fa560, 0x440b00)
	/Users/edd/Development/golang/src/github.com/influxdata/influxdb/cmd/influxd/run/server.go:216 +0xeb
github.com/influxdata/influxdb/monitor.(*Monitor).gatherStatistics(0xc4201800f0, 0xc44c738128, 0x1, 0x1, 0xc420261200, 0x0, 0x0, 0x0)
	/Users/edd/Development/golang/src/github.com/influxdata/influxdb/monitor/service.go:318 +0xd3
github.com/influxdata/influxdb/monitor.(*Monitor).Statistics(0xc4201800f0, 0xc420261200, 0xf76e80, 0x2540be400, 0xed0622da0, 0xc400000000, 0xf76e80)
	/Users/edd/Development/golang/src/github.com/influxdata/influxdb/monitor/service.go:310 +0xe65
github.com/influxdata/influxdb/monitor.(*Monitor).storeStatistics(0xc4201800f0)
	/Users/edd/Development/golang/src/github.com/influxdata/influxdb/monitor/service.go:411 +0x401
created by github.com/influxdata/influxdb/monitor.(*Monitor).Open
	/Users/edd/Development/golang/src/github.com/influxdata/influxdb/monitor/service.go:122 +0x2f8
ubuntu@ip-172-30-2-55:~$
```

**Comment by jwilder** [comment]
I also get this when dropping a DB w/ 400k series.

**Comment by e-dard** [comment]
@jwilder @benbjohnson all the tests are passing except for `TestShard_WritePoints_FieldConflictConcurrentQuery` which is timing out. I'm inclined to suggest we merge this, and get the fix for `TestShard_WritePoints_FieldConflictConcurrentQuery` in.

**Comment by jwilder** [comment]
Fixed via #8173
